core/storage/user/gae_models.py

```diff
@@ -131,6 +131,10 @@ class UserSettingsModel(base_models.BaseModel):
         repeated=True, indexed=True, choices=feconf.ALLOWED_USER_ROLES)
     # Flag to indicate whether the user is banned.
     banned = datastore_services.BooleanProperty(indexed=True, default=False)
+    # Flag to check whether the user has viewed lesson info modal once which
+    # shows the progress of the user through exploration checkpoints.
+    user_has_viewed_lesson_info_modal_once = datastore_services.BooleanProperty(
+        indexed=True, default=False)
```

Comment:
```suggestion
    has_viewed_lesson_info_modal_once = datastore_services.BooleanProperty(
        indexed=True, default=False)
```

The "user" is a bit redundant as this is already a user setting model.

Also, for fields like these we usually store the datetime (for example when the user started the tutorial `last_started_state_editor_tutorial`) would that be useful here?

Reply:
Updated the variable name now. 

In this case, we do not need to store the time user viewed the lesson info modal, since the property is simply suggestive of the fact that the user has viewed the lesson info once, and he now knows that he can check his checkpoint progress by opening the lesson info tab.

Reply:
In the other fields we also only check if the user has viewed something, I think the datatime filed is better in cases when we want to convey some new information in the lesson info and want to display it to all users again, if we used bool we would have to use a migration if we use datetime we only need to set new datetime that should be checked against.